Solution/WorkingFilesList.ToolWindow.Test/ViewModel/DocumentMetadataManagerTests.cs

```diff
@@ -2294,5 +2294,236 @@ public void ClearRemovesAllMetadata()
+            {
+                manager.TogglePinnedStatus(d);
+            }
+            manager.ActiveDocumentMetadata.Refresh();
+
+            return manager;
+        }
+
+        [Test]
+        public void FilterMatchesTwoDocuments()
+        {
+            // Arrange
+            const string document1Name = "AAADocument1";
+            const string document2Name = "BBBDocument2";
+            const string document3Name = "AABDocument3";
+
+            var manager = PrepareDocumentsForFilterTest(document1Name, document2Name, document3Name);
+
+            // Act
+
+            manager.FilterString = "BDoc";
+
+            var collection =
+                (IList<DocumentMetadata>)manager.ActiveDocumentMetadata.SourceCollection;
+
+            // Assert
+
+            var document1 = collection.Single(m => m.FullName == document1Name);
+            Assert.IsFalse(manager.ActiveDocumentMetadata.Contains(document1));
+            Assert.IsFalse(manager.PinnedDocumentMetadata.Contains(document1));
+
+            var document2 = collection.Single(m => m.FullName == document2Name);
+            Assert.IsTrue(manager.ActiveDocumentMetadata.Contains(document2));
+            Assert.IsTrue(manager.PinnedDocumentMetadata.Contains(document2));
+
+            var document3 = collection.Single(m => m.FullName == document3Name);
+            Assert.IsTrue(manager.ActiveDocumentMetadata.Contains(document3));
+            Assert.IsTrue(manager.PinnedDocumentMetadata.Contains(document3));
+        }
+
+        [Test]
+        public void FilterMatchesNoDocuments()
+        {
+            // Arrange
+            const string document1Name = "AAADocument1";
+            const string document2Name = "BBBDocument2";
+            const string document3Name = "AABDocument3";
+
+            var manager = PrepareDocumentsForFilterTest(document1Name, document2Name, document3Name);
+
+            // Act
+
+            manager.FilterString = "XXX";
+
+            var collection =
+                (IList<DocumentMetadata>)manager.ActiveDocumentMetadata.SourceCollection;
+
+            // Assert
+
+            var document1 = collection.Single(m => m.FullName == document1Name);
+            Assert.IsFalse(manager.ActiveDocumentMetadata.Contains(document1));
+            Assert.IsFalse(manager.PinnedDocumentMetadata.Contains(document1));
+
+            var document2 = collection.Single(m => m.FullName == document2Name);
+            Assert.IsFalse(manager.ActiveDocumentMetadata.Contains(document2));
+            Assert.IsFalse(manager.PinnedDocumentMetadata.Contains(document2));
+
+            var document3 = collection.Single(m => m.FullName == document3Name);
+            Assert.IsFalse(manager.ActiveDocumentMetadata.Contains(document3));
+            Assert.IsFalse(manager.PinnedDocumentMetadata.Contains(document3));
+        }
+
+
+        [Test]
+        public void FilterIsEmpytShowingAllDocuments()
```

Comment:
Small typo: `Empyt`

Would also suggest turning this into a parameterised test, i.e. `[TestCase()]`, with cases for `""`, `null`, and whitespace